### PR TITLE
Feature/test_util

### PR DESCRIPTION
**Pull request by kriii**
Add util tests and fix relevant it.

Changes about `ecVerifySig` and `ecRecoverPub` are crucial because `coincurve` doesn't support a verifying method with a recoverable signature, so converting it to a DER-encoded signature is needed. Other changes are minor.

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+from unittest import TestCase
```
Comment (liayoo) [inline tests/test_util.py:1]: `test_util.py` made me think that this file contains utils for the tests. Could you change the filename to something like `test_ain_utils.py`?

File: tests/test_util.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        self.assertFalse(isValidPrivate(bytes.fromhex(tmp)))
+
+    def testIsValidPrivateInvalidCurveEqN(self):
+        self.assertFalse(isValidPrivate(self.SECP256K1_N.to_bytes(32, "big")))
+    
+    def testIsValidPrivateInvalidCurveGtN(self):
+        self.assertFalse(isValidPrivate((self.SECP256K1_N + 1).to_bytes(32, "big")))
+
+    def testIsValidPrivateWork(self):
+        self.assertTrue(isValidPrivate((self.SECP256K1_N - 1).to_bytes(32, "big")))
```
Comment (liayoo) [inline tests/test_util.py:64]: question) why do we need the `-1` here?

File: tests/test_util.py (hunk 1 above)
Comment (liayoo) [reply tests/test_util.py:1]: Also, it would be very helpful if there was a short readme on how to run the test!

File: tests/test_util.py (hunk 2 above)
Comment (kriii) [reply tests/test_util.py]: That is to check working fine when the private key is less than N.

File: tests/test_util.py (hunk 1 above)
Comment (kriii) [reply tests/test_util.py]: Done!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

File: tests/test_ain_utils.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    def testIsValidPrivateInvalidCurveEqN(self):
+        self.assertFalse(isValidPrivate(self.SECP256K1_N.to_bytes(32, "big")))
+    
+    def testIsValidPrivateInvalidCurveGtN(self):
+        self.assertFalse(isValidPrivate((self.SECP256K1_N + 1).to_bytes(32, "big")))
+
+    def testIsValidPrivateWork(self):
+        self.assertTrue(isValidPrivate((self.SECP256K1_N - 1).to_bytes(32, "big")))
+
+# TODO(kriii): Add tests after implement `isValidPrivate`
```
Comment (platfowner) [inline tests/test_ain_utils.py:66]: minor: Could you add period (.) at the end of not only this one but all the TODO instances?
https://www.notion.so/comcom/ComCom-Developer-s-Guide-ad57ebc31ef94040a4f6e71d9c2042c5

File: ain/utils.py [hunk 4]
```diff
@@ -100,7 +101,7 @@ def toBytes(input: Any) -> bytes:
     elif type(input) is int:
         input = bytes.fromhex(padToEven(stripHexPrefix(hex(input))))
     elif input is None:
-        input = bytes([0])
```
Comment (minho-comcom-ai) [inline ain/utils.py:103]:
```python
>>> bytes([0])
b'\x00'
>>> bytes(0)
b''
```

Comment (minho-comcom-ai) [reply ain/utils.py:103]: Is `toBytes` equivalent and interoperable with `toBuffer` right?
([@ainblockchain/ain-util](https://github.com/ainblockchain/ain-util/blob/master/src/index.ts#L584-L608))
1.  Why don't we add this information on docstring area?
2. Now I understood that `bytes(0)` is right for `input is None` (`v === null || v === undefined`).

File: tests/test_ain_utils.py (hunk 3 above)
Comment (kriii) [reply tests/test_ain_utils.py]: I added all periods. Sorry!

File: ain/utils.py (hunk 4 above)
Comment (kriii) [reply ain/utils.py:103]: I fixed the docstring. Did you mean this right? If not, please comment. Thanks!

**Comment by kriii** [comment]
Merged. Thanks for the reviews!

File: ain/utils.py (hunk 4 above)
Comment (minho-comcom-ai) [reply ain/utils.py:103]: +1 @kriii

**Pull request merged**